### DISCUSSION TITLE
2.4.50 Upgrade - Missing Items/Elasticsearch Pending

### DISCUSSION THREAD

**Discussion by arisentow** [Security-Onion-Solutions/securityonion#12397] [Unsupported Versions]
I upgraded my 2.4.40 standalone to 2.4.50 via soup today and am not seeing the new features listed on the latest blog post, to include:
- "Connections and Metadata with community_id" and "Connections seen by Zeek or Suricata" dashboards
- Process Ancestors actions (when working with something process related)

Additionally, my previous 2.4.40 installation had the issue where elasticsearch was always "pending". It was temporarily fixed with #12302  however the issue came back after a few days. I was hoping it would be fixed with 2.4.50, however after the upgrade it is still there.

Any ideas?

**Comment by cm-ops** [comment]
Did that index roll over and create a replica shard again?

**Reply by arisentow** [reply]
It did on at least two occasions. If it helps, it has been the ".ds-.logs-endpoint.diagnostic.collection-default-". The latest one is as of last week. I haven't tried the fix since I upgraded to 2.4.50 yesterday, but will give it a shot and see if it comes back.

**Reply by arisentow** [reply]
Of note... the "fix" listed on the 2.4.50 release notes does not work. Nothing is listed if I run `sudo so-elasticsearch-indices-list | grep -vE "green|health"`. However, the commands in #12302 do.

**Reply by geistchevalier** [reply]
I'm getting the same issue, I'm still getting `UNASSIGNED` index a day after upgrading to 2.4.50, there weren't any replicas before upgrading.

**Reply by cm-ops** [reply]
A template was added in 2.4.50 https://github.com/Security-Onion-Solutions/securityonion/issues/12256, if it is still creating a replica it is probably still using the default Elasticsearch template.

**Reply by arisentow** [reply]
Should a fresh install of 2.4.50 be doing this? I did a clean install and sure enough two days later getting the "pending".

**Reply by weslambert** [reply]
Try adding a period to the start of the index pattern for the diagnostic template in `/opt/so/saltstack/default/salt/elasticsearch/defaults.yaml`

https://github.com/Security-Onion-Solutions/securityonion/blob/2.4/dev/salt/elasticsearch/defaults.yaml#L3929

**Reply by arisentow** [reply]
Do I need to run the fix first or anything? Also, after adding the period, do I run a salt-call?

**Reply by sleepingbel** [reply]
Hello Wes i had the same problem after a reboot today, but if i check the defaults.yaml it is the same as in github.
![Knipsel](https://github.com/Security-Onion-Solutions/securityonion/assets/29328171/bf82ea2b-e378-4332-a242-8c0c87f08142)

**Reply by weslambert** [reply]
The period change I mentioned will be in the next release. The template in 2.4.50 was assumed to fix the issue, but it didn't. 

You'll want to make the change to the defaults file with the period, then restart Elasticsearch. 

You can then check the template to see if it has been updated. The change to the replicas only affects indices that are already created. Assuming the template is updated, new indices should use 0 replicas.

**Reply by arisentow** [reply]
@weslambert I made the change above, and am still seeing "pending". Do I need to run one of the fixes to remove to old replicas? Also, while elasticsearch is in the pending status, is it actually affecting any functionality? I haven't noticed any glaring issues.
